src/Access/AccessControlHandler.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Drupal\password_enhancements\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Session\SessionManagerInterface;
+
+/**
+ * Defines access checks for the password_enhancements module.
+ */
+class AccessControlHandler implements AccessControlHandlerInterface {
+
+  /**
+   * Session manager.
+   *
+   * @var \Drupal\Core\Session\SessionManagerInterface
+   */
+  protected $sessionManager;
+
+  /**
+   * Constructs the access control.
+   *
+   * @param \Drupal\Core\Session\SessionManagerInterface $session_manager
+   *   Session manager.
+   */
+  public function __construct(SessionManagerInterface $session_manager) {
+    $this->sessionManager = $session_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function hasPasswordChangeAccess(AccountInterface $current_user): AccessResultInterface {
+    $password_change_required = $this->sessionManager->getBag('attributes')
+      ->getBag()
+      ->get('password_enhancements_password_change_required');
+
+    return AccessResult::allowedIf($password_change_required);
```

Comment:
Should not this AccessResult be tagged with the $current_user object as a dependency?

Reply:
Not sure, also not sure how to do that, but this if for the current session, the actual user is not checked here.

Reply:
ok, let's leave it like this and we'll whether it cause any trouble in testing.